addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -335,6 +350,37 @@ var FileWidget = SearchableMediaWidget.extend({
             return this.search(this.$('.o_we_search').val() || '');
         }
     },
+    /**
+     * Opens the image optimize dialog for the given attachment.
+     *
+     * Hides the media dialog while the optimize dialog is open to avoid an
+     * overlap of modals.
+     *
+     * @private
+     * @param {object} attachment
+     * @param {boolean} isExisting: whether this is a new attachment that was
+     *  just uploaded, or an existing attachment
+     * @returns {Promise} resolved with the updated attachment object when the
+     *  optimize dialog is saved. Rejected if the dialog is otherwise closed.
+     */
+    _openImageOptimizeDialog: function (attachment, isExisting) {
+        var self = this;
+        var promise = new Promise(function (resolve, reject) {
+            self.trigger_up('hide');
+            new ImageOptimizeDialog(self, {}, {
+                attachment: attachment,
+                isExisting: isExisting,
+                optimizedWidth: self._computeOptimizedWidth(),
+                reject: reject,
+                resolve: resolve,
+            }).open();
+        });
+        var always = function () {
+            self.trigger_up('show');
```

Comment:
rename into show_parent_dialog_request or something like that, those are event name, not names of action to perform

Reply:
Ok much better. I wasn't sure if we were supposed to be concise or not. I hate all of those short and barely greppable `show`, `save`, etc...